sources/input_to_text_tree_mocks/mock_2.c

```diff
@@ -22,7 +22,7 @@ void	mock_2_tree(t_treenode *root)
 /* returns 1 on success - to allow condition usage outside */
 int	mock_2(char *input, t_treenode *root)
 {
-	if (ft_strcmp(input, MOCK_2_TEXT) != 0)
+	if (ft_strcmp(input, MOCK_2_TEXT) != 0 && ft_strcmp(input, "MOCK_2") != 0)
 		return (0);
 	mock_2_tree(root);
 	return (1);
```

Comment:
why do you return somewhere 0 on success and somewhere 1?

Reply:
That's not success. Succes of failure is stored in retval. This 0/1 here means that the function got to the end or got terminated early. That's like `found` variable in command chooser.